web/src/user_profile.ts

```diff
@@ -706,26 +722,32 @@ export function show_user_profile(user: User, default_tab_key = "profile-tab"):
                     break;
                 }
                 case "user-profile-streams-tab": {
-                    if (!user_streams_list_widget) {
-                        const user_streams = stream_data.get_streams_for_user(
-                            user.user_id,
-                        ).subscribed;
-                        if (show_user_subscribe_widget) {
-                            render_user_profile_subscribe_widget();
+                    void (async () => {
+                        if (!user_streams_list_widget) {
+                            const user_streams = (
+                                await stream_data.get_streams_for_user(user.user_id)
```

Comment:
I think making `show_user_profile` async is the wrong approach; the more natural model would be that opening that tab immediately shows a loading spinner and triggers the API request to get the data and hide the loading spinner, with a callback to render the list once it returns. That shouldn't require this function itself to itself use async/await?

Reply:
If we can remove the async call from the discussion above then we can remove the async on `show_user_profile`. But adding a loading spinner here is probably still a good idea.